### PR TITLE
Add spade support

### PR DESCRIPTION
**Pull request by TheZoq2**
Adds support for https://spade-lang.org/

### CODE REVIEW

File: languages.toml [hunk 1]
```diff
@@ -3726,3 +3727,29 @@ grammar = "typescript"
...
+]
+language-servers = [ "spade-language-server" ]
+indent = { tab-width = 4, unit = "    " }
+
+[language.auto-pairs] 
+'(' = ')' 
+'{' = '}' 
+'[' = ']' 
+'"' = '"' 
+'<' = '>'
```
Comment (TheZoq2) [inline languages.toml:3751]: This doesn't seem to work and I can't figure out why, is it something wrong with the config?

Comment (TheZoq2) [reply languages.toml:3751]: This was caused by me using my local config which overrode the one I tried adding to the runtime, should be fixed now

File: runtime/queries/spade/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+((op_bitwise_and) @operator)
+((op_bitwise_xor) @operator)
+((op_bitwise_or) @operator)
+((op_logical_and) @operator)
+((op_logical_or) @operator)
+
+
+[
+  (line_comment)
+] @comment @spell
```
Comment (the-mikedavis) [inline runtime/queries/spade/highlights.scm:109]:
```suggestion
] @comment
```

`@spell` is specific to neovim

File: runtime/queries/spade/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+((identifier) @type
+  (#match? @type "[A-Z]"))
+
+((scoped_identifier
+    name: (identifier) @type)
+ (#match? @type "^[A-Z]"))
+
+((identifier) @constant
+ (#match? @constant "^[A-Z][A-Z%d_]*$"))
```
Comment (the-mikedavis) [inline runtime/queries/spade/highlights.scm:120]:
```suggestion
 (#match? @constant "^[A-Z][A-Z\\d_]*$"))
```

I believe the `%` is used in `#lua-match?` from neovim, here it needs a double escape

**Review by the-mikedavis — COMMENTED** [review]
To fix the docs CI you can run `cargo xtask docgen` and commit the changes

**Review by the-mikedavis — APPROVED** [review]

**Comment by TheZoq2** [comment]
Pushed a small update to follow the same rationale as #8535

**Review by the-mikedavis — DISMISSED** [review]

**Comment by TheZoq2** [comment]
Pushed another small update to bump the grammar and add special comments

**Review by the-mikedavis — APPROVED** [review]

**Pull request merged**